src/filters/events.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const { now, getDate } = require('./time');
+
+/* @docs
+label: Event Filters
+category: File
+*/
+
+/* @docs
+label: buildEvent
+category: List
+note: |
+  Build an event-page out of combined event and page data,
+  so that events can be treated as a page of their own
+params:
+  page:
+    type: page object
+  event:
+    type: object
+*/
+const buildEvent = (page, event) => ({
+  date: event ? getDate(event.date) || page.date : page.date,
```

Comment:
Is `event` an optional argument?  If `buildEvent` will always be called with an `event` and we intend to check if the event has a date, it may make sense to write it like this:

```suggestion
  date: event.date ? getDate(event.date)  : page.date,
```